### PR TITLE
Feature/94 remote

### LINKED ISSUE #94: Check if pytrip can connect to trip
Starting TRiP98 from pytrip might not work now, paramiko is not on a list of requirements.
Tests to connect to TRiP98 remotely are missing.

### PR DESCRIPTION
**Pull request by nbassler**
- repairs remote execution, fixing #94 
- new: passwordless login using RSA keys now possible
- new: stdout and stdin TRiP execution may be dumped into user specified files
- new: user may set a specific dir for remote execution
- more doctrings #91 
- removed `create_remote_run_file()` function (which was internal anyway)

### CODE REVIEW

**Comment by nbassler** [comment]
@grzanka regarding tests: I did not add any
- remote tests needs login credentials or secret key
- remote TRiP installation needed too, but I have added an override for future dry runs. Simply set `tripexecutor._runtrip = False` prior execution

File: pytrip/tripexecuter/tripexecuter.py [hunk 1]
```diff
@@ -670,27 +764,48 @@ def convert_files_to_voxelplan(self):
...
+        # """
+        # Generates script called 'run.sh' which sources ~/.profile and runs 'TRiP98 < plan.exec'
+        # """
+        # logger.debug("Generate 'run.sh' script in " + self.working_path)
+        # with open(os.path.join(self.working_path, self.folder_name, "run.sh"), "wb+") as fp:
+        #     fp.write("#!/bin/bash -lc")
+        #     # fp.write("source ~/.profile\n")
+        #    # fp.write("cat ~/.bashrc > foo.log\n")
+        #    fp.write("TRiP98 < plan.exec > plan.log 2> planerr.log\n")
+        #    fp.close()
```
Comment (grzanka) [inline pytrip/tripexecuter/tripexecuter.py]: Can be removed and retrieved from GIT history.

File: pytrip/tripexecuter/tripexecuter.py [hunk 2]
```diff
@@ -670,27 +764,48 @@ def convert_files_to_voxelplan(self):
...
+            path1 = path.replace(".exec", "")
+            for projectile in self.projectiles:
+                self.create_trip_exec(projectile, True)
+                with open(path1 + "_" + projectile + ".exec", "wb+") as fp:
+                    fp.write(self.trip_exec)
+                    fp.close()
+        else:
+            with open(path, "wb+") as fp:
                 fp.write(self.trip_exec)
+                fp.close()
```
Comment (grzanka) [inline pytrip/tripexecuter/tripexecuter.py]: closing not needed, "with + open" does it automatically

File: pytrip/tripexecuter/tripexecuter.py [hunk 3]
```diff
@@ -569,38 +592,109 @@ def run_trip(self):
...
-            if (retcode is not None):
-                break
+            answer_stdout = p.stdout.readline()
+            answer_stderr = p.stderr.readline()
+            logger.debug("Remote answer stdout:" + answer_stdout)
+            logger.debug("Remote answer stderr:" + answer_stderr)
+            fp_stdout.write(answer_stdout)
+            fp_stderr.write(answer_stderr)
+            self.log(answer_stdout)
+            if (retcode is not None):  # runs until process stops
```
Comment (grzanka) [inline pytrip/tripexecuter/tripexecuter.py]: brackets not needed

**Pull request merged**